### ISSUE TITLE
UI/API: Not showing all my VMs as normal users when projects is selected 

### ISSUE THREAD

**Issue by rohityadavcloud** [apache/cloudstack#7642]
As a normal user when I select projects toggle switch in the VM view, I only see project VMs but don't see my own VMs. However, as a root admin the behaviour is different wherein selecting projects shows me both all VMs and all VMs across projects.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI, API
~~~

##### CLOUDSTACK VERSION

~~~
4.18.0.0
~~~

##### STEPS TO REPRODUCE

~~~
1. Deploy VM for a normal user account and also in a project.
2. Try to see VMs list in the VM view you either see you own VMs or projects.
~~~
Screenshots of the VM view as normal user with/without the projects switch enabled:

<img width="1408" alt="Screenshot 2023-06-16 at 9 59 23 PM" src="https://github.com/apache/cloudstack/assets/95203/05531567-7a44-4cd8-a797-287097ecbaa4">
<img width="1412" alt="Screenshot 2023-06-16 at 9 59 30 PM" src="https://github.com/apache/cloudstack/assets/95203/7f9eda51-e8c8-46b1-a45c-b5c25e0b40ee">

##### EXPECTED RESULTS

~~~
When projects switch is on in the UI, I expect to see both my own VMs and from my projects.
~~~

**Comment by DaanHoogland** [comment]
For both user and admin the exact same set of calls is done:
~~~
http://10.0.35.7:8080/client/api/?listall=true&projectid=-1&page=1&pagesize=20&showIcon=true&command=listVirtualMachinesMetrics&response=json
http://10.0.35.7:8080/client/api/?listall=true&page=1&pagesize=20&showIcon=true&command=listVirtualMachinesMetrics&response=json
~~~
This means it is an API issue. I will investigate if this is easily fixable. I do recall mailing (or github) discussions about the subject and will try to find those. It might be intentional , i.e. not a bug @rohityadavcloud but your intention is clear.

**Comment by DaanHoogland** [comment]
Note @rohityadavcloud that I see this behaviour also for networks for admin as well as notmal users. I think we need a consistent functional design for this. 
a proposal would be `listall=true projectid=-1` should always show everything of the requested resource type the user has access to. This would probably mean a lot of changes. At least more than the one I attempted so far.

**Comment by rohityadavcloud** [comment]
Agree with you suggestion @DaanHoogland I would keep the scope limited and iterate, and also watch out for any potential side effects.

**Comment by DaanHoogland** [comment]
fixed in #7667
